packages/api/src/routes/medical/internal-docs.ts

```diff
@@ -303,4 +305,44 @@ router.post(
   })
 );
 
+/**
+ * POST /internal/docs/query
+ *
+ * Starts a new document query even if the current one is in 'processing' state.
+ * @param req.query.cxId - The customer/account's ID.
+ * @param req.query.patientId - The customer/account's ID.
+ * @param req.query.facilityId - Optional; The facility providing NPI for the document query.
+ *
+ * @return updated document query progress
+ */
+router.post(
+  "/query",
+  asyncHandler(async (req: Request, res: Response) => {
+    const cxId = getFromQueryOrFail("cxId", req);
+    const patientId = getFromQueryOrFail("patientId", req);
+    const facilityId = getFrom("query").optional("facilityId", req);
+
+    const patient = await getPatientOrFail({ cxId, id: patientId });
+    const patientUpdate: PatientUpdateCmd = {
+      ...patient.data,
+      id: patient.id,
+      cxId: patient.cxId,
+    };
+
+    if (patientUpdate.documentQueryProgress) {
+      patientUpdate.documentQueryProgress = undefined;
+    }
+
+    await updatePatient(patientUpdate, true);
```

Comment:
Let's use `appendDocQueryProgress()` instead, please.

Usually, we don't care too much about concurrency while updating a DB record, but in this case we do. There's a lot of concurrent updates on Patient's doc query progress object, especially WRT the conversion updates.

That function is safe:
- it starts a DB tx;
- loads the patient with a row lock (so it can't be updated while the DB tx lasts;
- updates the patient;
- commits the DB dx.

In this case I think we'd call it like so, just like we do when a doc query is triggered by the cx, on [processDocuments()](https://github.com/metriport/metriport/blob/ce31b7d78316a9921ac46d52674be76c25b027e1/packages/api/src/command/medical/document/document-redownload.ts#L214):

```ts
    await appendDocQueryProgress({
      patient: { id: patient.id, cxId: patient.cxId },
      downloadProgress: { status: "processing" },
      reset: true,
    });
```